ci(flaky): test_no_host_paths times out (>120s) under runner I/O stall, aborts whole suite via -x

## Summary

`tests/test_no_host_paths.py::test_no_host_specific_absolute_paths` intermittently fails CI with `Failed: Timeout (>120.0s) from pytest-timeout`, even though the test completes in **~0.4s** in isolation. Because the suite runs with `-x` (fail-fast), this single transient timeout aborts the **entire** `call-test-lint` job and red-flags otherwise-green PRs.

Observed on PR #755 (`call-test-lint / Test and Lint`, run `28295440326`): every other check passed; the only failure was this timeout. A failed-jobs re-run was triggered to unblock — but the flake will recur.

## Evidence

CI failure (truncated):

```
=================================== FAILURES ===================================
_____________________ test_no_host_specific_absolute_paths _____________________
...
>               text = path.read_text(encoding="utf-8")
tests/test_no_host_paths.py:89:
...
E   Failed: Timeout (>120.0s) from pytest-timeout.
```

Local timing of the same test, same commit:

```
tests/test_no_host_paths.py .                                            [100%]
============================== 1 passed in 0.40s ===============================
0:00.75elapsed
```

The test is a plain regex sweep over a few hundred small `.py` files. There is no operation in it that can legitimately take 120s. The traceback shows it blocked inside `f.read()` on a small file — consistent with a transient runner I/O stall, not a code defect.

## Root cause (mechanism)

In `pyproject.toml`:

```toml
[tool.pytest.ini_options]
addopts = "-v --cov=strands_robots --cov-report=term-missing --strict-markers --timeout=120"
```

and `.github/workflows/test-lint.yml` runs `hatch run test -x ...`.

- `--timeout=120` is a single **global per-test** budget. pytest-timeout's default method measures full wall-clock (including fixture setup and coverage instrumentation), so under runner I/O contention any one test can momentarily exceed it.
- `-x` makes that one transient timeout abort the whole suite.

So a millisecond-scale hygiene test becomes a full-CI failure whenever the runner hiccups. This gets worse as the repo grows (more files to walk, larger coverage flushes).

## Proposed fix (smallest viable)

Decouple a transient stall from a hard suite failure. Options, in order of preference:

1. **Pin a generous timeout marker on the I/O-bound sweep test** rather than relying on the global 120s — e.g. mark `test_no_host_paths.py` with `@pytest.mark.timeout(0)` (disable) or a large value, since it is a deterministic hygiene check whose slowness is purely environmental, not algorithmic. This is the most surgical change.
2. Raise the global `--timeout` modestly (e.g. 300) to absorb runner I/O jitter without masking genuine hangs.

Option 1 is preferred: it is targeted, documents intent at the test, and leaves the strict 120s budget protecting the rest of the suite.

## Acceptance criteria

- The host-paths hygiene sweep cannot fail CI due to a transient runner I/O stall.
- Genuine hangs elsewhere in the suite still trip the 120s budget.
- No change to what the test actually validates (it must still catch `/Users/<name>/`, `/home/<name>/`, `C:\Users\<name>\`).

## Notes

Intentionally filed as a standalone issue rather than folded into PR #755 (an approved, scope-tight hardware feature PR) to keep that diff coherent.

--- a/tests/test_no_host_paths.py
+++ b/tests/test_no_host_paths.py
@@ -18,6 +18,8 @@ from __future__ import annotations
 import re
 from pathlib import Path
 
+import pytest
+
 REPO_ROOT = Path(__file__).resolve().parent.parent
 
 # Directories to scan (source + tests; not docs, not third-party).
@@ -66,6 +68,15 @@ def _iter_source_files() -> list[Path]:
     return files
 
 
+# This sweep walks a few hundred small .py files and completes in well under a
+# second. Its only failure mode under the global ``--timeout=120`` budget is a
+# transient runner I/O stall on ``Path.read_text`` - an environmental hiccup,
+# not an algorithmic hang. With the suite running fail-fast (``-x``), one such
+# stall aborts the entire job and red-flags otherwise-green PRs. Disable the
+# per-test timeout here (``timeout(0)``) so this deterministic hygiene check is
+# never governed by the wall-clock budget; the strict 120s budget still
+# protects every other test from genuine hangs.
+@pytest.mark.timeout(0)
 def test_no_host_specific_absolute_paths() -> None:
     """Fail if any .py file contains ``/Users/<name>/`` or ``/home/<name>/``.
 
@@ -101,3 +112,19 @@ def test_no_host_specific_absolute_paths() -> None:
         for rel, lineno, snippet in offenders:
             msg.append(f"  {rel}:{lineno}: {snippet}")
         raise AssertionError("\n".join(msg))
+
+
+def test_host_path_sweep_disables_global_timeout() -> None:
+    """Guard the flake fix: the sweep must opt out of the global per-test timeout.
+
+    ``test_no_host_specific_absolute_paths`` is a deterministic, sub-second regex
+    sweep whose only way to exceed the global ``--timeout=120`` budget is a
+    transient runner I/O stall. Under fail-fast (``-x``), one such stall aborts
+    the whole suite. We pin ``@pytest.mark.timeout(0)`` so the wall-clock budget
+    cannot govern it. This regression asserts that opt-out stays in place; it
+    fails if the marker is dropped or set to a finite budget.
+    """
+    pytestmark = getattr(test_no_host_specific_absolute_paths, "pytestmark", [])
+    marks = [m for m in pytestmark if m.name == "timeout"]
+    assert marks, "expected a @pytest.mark.timeout marker on the host-path sweep"
+    assert marks[0].args == (0,), f"expected timeout(0) to disable the budget, got {marks[0].args!r}"

